### ISSUE TITLE
Browser.sh location

### ISSUE THREAD

**Issue by CalvinRodo** [devcontainers/features#463]
Hey folks, I'm trying to create a feature for a tool that needs to call out to the browser, unfortunately the `BROWSER` env doesn't seem be to set when calling install.sh and also I can't find the browser.sh file in the filesystem where I would normally locate it in a devcontainer, do you have any suggestions for what i can do?

**Comment by samruddhikhandale** [comment]
Hi 👋 

I wonder if your Feature could add a `browser` option that would be accessible to the `install.sh` file? 
The user can add your feature like 👇 

``` jsonc
// devcontainer.json
"features": {
    "ghcr.io/owner/repo/my-feature:1": {
        "browser": "/usr/bin/x"
    }
}

```

https://containers.dev/implementors/features/#options-property

@CalvinRodo Would that help?

**Comment by CalvinRodo** [comment]
Not really, by default we don't have access to the underlying filesystem (afaik) so I'm assuming we would have to couple that with a mount in the devcontainer to expose that file, which would be less then ideal. 

I really want to get access to the shellcript that sits in the $BROWSER environment variable. 

For instance in a dev-container I'm running now that is:

```
BROWSER="/vscode/vscode-server/bin/linux-x64/441438abd1ac652551dbe4d408dfcec8a499b8bf/bin/helpers/browser.sh"
```

**Comment by jkeech** [comment]
The `$BROWSER` variable is set only in VS Code interactive terminals when connected to a remote environment. Since it's VS Code specific and only present once VS Code is connected to the container, it's not visible to feature install scripts that run when building the container image.

What is your feature trying to do with the $BROWSER at image build time? Perhaps there's another way to structure it to do what you need.

**Comment by CalvinRodo** [comment]
So what I was trying to do was write a feature for the granted cli tool.

https://www.granted.dev/

One of the things it can do is spawn a browser for cloud console access. However I was able to just solve my issue using a docker container, and telling granted to use the browser.sh using a postCrwateCommand script.

Looks like the fix will have to be in the granted tool and not in dev containers.

Thanks for following up.

**Comment by CalvinRodo** [comment]
Please feel free to close this as not possible.

**Comment by jkeech** [comment]
Sounds good. You could definitely install the `granted` tool using the feature install script and then spawn `granted` at container runtime inside a VS Code interactive terminal which has `$BROWSER` set 👍. Features have the ability to [contribute lifecycle hooks](https://github.com/devcontainers/spec/blob/main/proposals/features-contribute-lifecycle-scripts.md) now which might help. I could see this being useful as `postAttachCommand` contributed by the feature. Note that in GitHub Codespaces, `postAttachCommands` contributed by features don't work yet, but that should be coming in the future.
